chains/evm/contracts/pools/USDC/CCTPTokenPool.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {IPoolV1} from "../../interfaces/IPool.sol";
+import {IPoolV2} from "../../interfaces/IPoolV2.sol";
+import {ITypeAndVersion} from "@chainlink/contracts/src/v0.8/shared/interfaces/ITypeAndVersion.sol";
+
+import {Pool} from "../../libraries/Pool.sol";
+import {USDCSourcePoolDataCodec} from "../../libraries/USDCSourcePoolDataCodec.sol";
+import {TokenPool} from "../TokenPool.sol";
+import {AuthorizedCallers} from "@chainlink/contracts/src/v0.8/shared/access/AuthorizedCallers.sol";
+
+import {IERC20} from "@openzeppelin/contracts@4.8.3/token/ERC20/IERC20.sol";
+
+/// @notice CCTP token pool that delegates minting and burning responsibilities of USDC to the CCTPVerifier contract.
+/// @dev This pool does not mutate the token state. It does not actually burn USDC via TokenMessenger on source or mint via MessageTransmitter on destination.
+/// It remains responsible for rate limiting and other validations while outsourcing token management to the CCTPVerfier contract.
+/// This token pool should never have a balance of USDC at any point during a transaction, otherwise funds will be lost.
+/// The caller of lockOrBurn is responsible for sending USDC to the CCTPVerifier contract instead.
+contract CCTPTokenPool is TokenPool, ITypeAndVersion, AuthorizedCallers {
+  string public constant override typeAndVersion = "CCTPTokenPool 1.7.0-dev";
+
+  constructor(
+    IERC20 token,
+    uint8 localTokenDecimals,
+    address advancedPoolHooks,
```

Comment:
Can you remove this input and pass `address(0)` to the TokenPool, and apply that to all USDC pools? And we can kill the hooks for it like other USDC pools 

```
  /// @notice No-op override to purge the unused code path from the contract.
  function _postFlightCheck(Pool.ReleaseOrMintInV1 calldata, uint256, uint16) internal pure virtual override {}

  /// @notice No-op override to purge the unused code path from the contract.
  function _preFlightCheck(Pool.LockOrBurnInV1 calldata, uint16, bytes memory) internal pure virtual override {}
```